### DISCUSSION TITLE
关于不同情形下选择对应profile的设想。

### DISCUSSION THREAD

**Discussion by sqq135** [hooke007/mpv_PlayKit#363] [mpv 通用]
平时看动漫，大多是1080p，也设置好合适的profile加载好滤镜染色器，但是偶尔看4k，性能就不太够，稍有点卡，时常忘记调回来。因此想根据横向分辨率是否大于1920选择不同profile。
求助于chatgpt，但是不行，以下是它给的代码，lua文件，放进script文件夹里。
请问该如何写这个脚本呢？或者根据不同文件夹选择脚本，也另外是个思路

-- 根据分辨率选择配置文件

-- 获取视频的横向分辨率
function getVideoWidth()
    local video_params = mp.get_property_native("video-out-params")
    return video_params and video_params["w"] or 0
end

-- 根据分辨率选择配置文件
function selectProfile()
    local width = getVideoWidth()
    if width > 1920 then
        mp.commandv("apply-profile", "sub-SRT")
    else
        mp.commandv("apply-profile", "AMK自动")
    end
end

-- 监听视频加载事件，并在加载时选择配置文件
mp.register_event("file-loaded", selectProfile)

**Comment by hooke007** [comment]
在滥用openai的产品之前，我的建议是首先熟悉mpv的基本用法。这种简单需求只是写两个profile的事情

**Reply by sqq135** [reply]
我就说哪里看过，时间太长忘记了，感谢指导，写出来了，profile-cond=p["video-params/w"]>1920
如果说想要写条件是 D盘的“视频”“movie”文件夹，应该怎么写啊，实在没有在命令接口的文件中找到。

https://hooke007.github.io/unofficial/mpv_profiles.html#profile

**Reply by hooke007** [reply]
> 判定的核心对象是 “属性” 即手册中的[属性列表](https://mpv.io/manual/master/#property-list)中列出的全部，

去属性列表里找涉及路径的属性。

而且官方文档里就有对应的示例
https://mpv.io/manual/master/#conditional-auto-profiles
